server/src/main/java/org/opensearch/wlm/WorkloadManagementTransportInterceptor.java

```diff
@@ -20,11 +21,11 @@
  */
 public class WorkloadManagementTransportInterceptor implements TransportInterceptor {
     private final ThreadPool threadPool;
-    private final QueryGroupService queryGroupService;
+    private final SetOnce<QueryGroupService> queryGroupService;
 
-    public WorkloadManagementTransportInterceptor(final ThreadPool threadPool, final QueryGroupService queryGroupService) {
+    public WorkloadManagementTransportInterceptor(final ThreadPool threadPool, final SetOnce<QueryGroupService> queryGroupServiceSetOnce) {
         this.threadPool = threadPool;
-        this.queryGroupService = queryGroupService;
+        this.queryGroupService = queryGroupServiceSetOnce;
```

Comment:
Why do we need `SetOnce` here instead of usual class reference? I am assuming there is single `WorkloadManagementTransportInterceptor` instance per node